pkg/ccl/backupccl/restore_online_test.go

```diff
@@ -26,6 +27,8 @@ func TestOnlineRestoreBasic(t *testing.T) {
 	const numAccounts = 1000
 	_, sqlDB, dir, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
 	defer cleanupFn()
+
+	defer nodelocal.HijackNodeLocalForTesting(t.TempDir())()
```

Comment:
The normal nodelocal storage provider isn't available at early boot because it requires that you have a node ID and can send RPCs to other nodes.  Oneline restore requires a storage provider that is available before those things are true since we may need to reboot and read from the store very early in startup.

We could have made a whole different storage provider just for testing. And that may be been a bit cleaner, but this allows us to replace the nodelocal storage provider with an implementation that just serves files out of a temp directory without adding a new provider.